### PR TITLE
feat(ITV Hub): rebrand ITV Hub to ITVX

### PR DESCRIPTION
**Pull request by hiitsdan**
## Description 
ITV recently released ITVX which completely replaces the previous ITV Hub. This PR updates the ITV Hub presence to support the full ITVX rebrand.

## Acknowledgements
- [x] I read the [Presence Guidelines](https://github.com/PreMiD/Presences/blob/main/.github/CONTRIBUTING.md)
- [x] I linted the code by running `yarn format`
- [x] The PR title follows the repo's [commit conventions](https://github.com/PreMiD/Presences/blob/main/.github/COMMIT_CONVENTION.md)

## Screenshots
<details>
<summary> Proof showing the creation/modification is working as expected </summary>

image
image
image
image

</details>

### CODE REVIEW

File: websites/I/ITVX/presence.ts [hunk 1]
```diff
@@ -0,0 +1,188 @@
...
+		const nextDataElement = document.querySelector("#__NEXT_DATA__");
+
+		if (!nextDataElement) return null;
+
+		return JSON.parse(nextDataElement.textContent);
+	};
+
+presence.on("UpdateData", async () => {
+	const presenceData: PresenceData = {
+			largeImageKey: "itv_logo",
```
Comment (theusaf) [inline websites/I/ITVX/presence.ts:82]: Imgur URLs are preferred over Discord asset names now, replace.
```suggestion
			largeImageKey: "https://i.imgur.com/XisKvdg.png",
```

File: websites/I/ITVX/presence.ts [hunk 2]
```diff
@@ -0,0 +1,188 @@
...
+							presence.getTimestamps(
+								Math.floor(video.currentTime),
+								Math.floor(video.duration)
+							);
+
+						presenceData.smallImageKey = "play";
+						presenceData.smallImageText = "Playing";
+					} else {
+						presenceData.smallImageKey = "pause";
+						presenceData.smallImageText = "Paused";
```
Comment (theusaf) [inline websites/I/ITVX/presence.ts:180]: Replace with Imgur URLs and use an `Assets` enum

File: websites/I/ITVX/presence.ts [hunk 3]
```diff
@@ -0,0 +1,188 @@
...
+
+		return JSON.parse(nextDataElement.textContent);
+	};
+
+presence.on("UpdateData", async () => {
+	const presenceData: PresenceData = {
+			largeImageKey: "itv_logo",
+			startTimestamp: Date.now(),
+		},
+		path = document.location.pathname,
```
Comment (theusaf) [inline websites/I/ITVX/presence.ts:85]:
```suggestion
		{ pathname: path } = document.location,
```
or
```suggestion
		{ pathname } = document.location,
```

**Review by theusaf — CHANGES_REQUESTED** [review]

File: websites/I/ITVX/presence.ts [hunk 4]
```diff
@@ -0,0 +1,195 @@
...
+			};
+		};
+	};
+}
+
+enum Assets {
+	logo = "https://i.imgur.com/XisKvdg.png",
+	live = "https://i.imgur.com/WTcQZS7.png",
+	paused = "https://i.imgur.com/T6jkHHy.png",
+	playing = "https://i.imgur.com/1CbifM8.png",
```
Comment (EncryptedDev) [inline websites/I/ITVX/presence.ts:59]: enum properties should be TitleCase

File: websites/I/ITVX/presence.ts [hunk 5]
```diff
@@ -0,0 +1,195 @@
...
+
+		return JSON.parse(nextDataElement.textContent);
+	},
+	fetchProgrammeNextData = (): ProgrammeNextData => {
+		const nextDataElement = document.querySelector("#__NEXT_DATA__");
+
+		if (!nextDataElement) return null;
+
+		return JSON.parse(nextDataElement.textContent);
+	};
```
Comment (EncryptedDev) [inline websites/I/ITVX/presence.ts:85]: These should be 'proper' functions invoked with the `function` keyword instead of variable functions

**Review by EncryptedDev — CHANGES_REQUESTED** [review]

File: websites/I/ITVX/presence.ts [hunk 6]
```diff
@@ -0,0 +1,198 @@
...
+	return JSON.parse(nextDataElement.textContent);
+}
+
+function fetchProgrammeNextData(): ProgrammeNextData {
+	const nextDataElement = document.querySelector("#__NEXT_DATA__");
+
+	if (!nextDataElement) return null;
+
+	return JSON.parse(nextDataElement.textContent);
+}
```
[the remark this answers is not in the document]
Comment (hiitsdan) [reply websites/I/ITVX/presence.ts]: You're right. Not overly familiar with TS so was not aware generics were a thing.

**Review by Bas950 — APPROVED** [review]

**Review by EncryptedDev — APPROVED** [review]

**Pull request merged**